Added new content  ( Dynamic starter kit guide ) (#160)
rsksmart/devportal: docs/02-developers/04-quickstart/Dynamic Starter Kit_ Accelerate User Onboarding on Rootstock.md

3. **Token Transfers**

Token transfers allow users to send assets to other addresses directly from the dApp. This feature lets users select a token, specify an amount, and input a recipient address to complete the transfer. With wagmi’s transaction hooks, token transfers are handled seamlessly, removing the need for complex contract interactions. This functionality is useful for dApps where peer-to-peer payments or transfers are common, like in DeFi or tipping applications.

4. **Message Signing and Verification**

Message signing lets users prove their identity or authorize actions without exposing sensitive information. This feature allows users to sign and verify messages, which is useful for secure authentication or transaction confirmation. Wagmi’s signature hooks streamlines the signing process, providing both security and flexibility for the dApp. It’s especially helpful for applications where identity verification is required.

5. **Rootstock Testnet Support**

> This project is preconfigured for the **Rootstock Testnet**, which allows developers to test Web3 features without spending real assets. Rootstock is an EVM-compatible blockchain with the security of Bitcoin, offering a realistic environment for development. By building on the testnet, developers can ensure their dApp is ready for deployment to the mainnet. This provides a risk-free space to experiment with blockchain features, making it ideal for early-stage development and testing.

Reviewer: ```suggestion
This project is preconfigured for the **Rootstock Testnet**, which allows developers to test dApps without spending real assets. By building on the testnet, developers can ensure their dApp is ready for deployment to the mainnet. This provides a risk-free space to experiment with blockchain features, making it ideal for early-stage development and testing.
```

## **Prerequisites**

> This project leverages some key libraries to handle server-side rendering, Web3 interactions, and blockchain contract communication. 

Reviewer: ```suggestion
This project leverages key libraries to handle server-side rendering, Web3 interactions, and blockchain contract communication. 
```

Before starting the project, make sure you have these essential tools installed on your computer:

1. **Node.js**:  
   * You’ll need Node.js, version **19.x** or later. Node.js allows you to run JavaScript on the server, which is required for building and running modern web applications.  
     [Download Node.js here](https://nodejs.org/) if you haven't installed it yet.  
2. **Bun** or **Yarn** (recommended for Next.js projects):  
   * **Bun** (version **1.1.x** or later): A fast JavaScript runtime and package manager.  
     See how to [Download Bun](https://bun.sh/).

3. **Next.js**:

   * Next.js is a powerful React framework that enables server-rendered web applications, helping to make your website faster and more SEO-friendly.  
    In this project, Next.js serves as the backbone for building the front end.

4. **Wagmi**:

   * `wagmi` is a collection of React hooks for interacting with Web3, which lets you connect to blockchain networks, handle user authentication, and more.  
   This library makes it easier to integrate Web3 functionality into React components.

5. **Viem**:

>     * `viem` provides an easy way  to interact with smart contracts on the Rootstock blockchain.  

Reviewer: ```suggestion
    * `viem` provides an easy way to interact with smart contracts on the Rootstock blockchain.  
```

    This library will be used to connect to Rootstock and make contract calls.

:::warning[Warning]

> This is a starter kit designed for hackathons and rapid prototyping. It is intended for educational and experimental purposes only. Use it at your own risk, and ensure thorough testing before deploying in production environments.

Reviewer: ```suggestion
This is a starter kit designed for rapid prototyping. It is intended for educational and experimental purposes only. Use it at your own risk, and ensure thorough testing before deploying in production environments.
```

:::

## **Getting Started**

<Steps>
  <Step title=" Clone the Repository">
   Clone the repository to use the starter kit locally.

> ```

Reviewer: ```suggestion
```bash
```

git clone https://github.com/RookieCol/rootstock-dynamic
cd rootstock-dynamic
```
  </Step>
  <Step title="Install Dependencies">
    Install the necessary dependencies with either Bun or Yarn.
  <Tabs>
  <TabItem value="contribute" label="Using Bun" default>
    ```
    bun install
    ```
  </TabItem>
  <TabItem value="contest" label="Using Yarn">
   ```
   yarn install
   ```
  </TabItem>

</Tabs>

  </Step>

>   <Step title="Get Environment Variables from Dynamics">

Reviewer: ```suggestion
  <Step title="Get Environment Variables from Dynamic">
```

  :::note
Create a FREE account on Dynamic and login to your Dashboard. Then obtain your `ENVIRONMENT_ID` from the [Dynamic dashboard](https://app.dynamic.xyz/dashboard/overview). 

:::
   
Follow these steps to locate and copy your Environment ID:

> <img src="/img/developers/quickstart/dynamic-starter-kit/image1.png" alt="Dynamic.xyz dashobard"/>

Reviewer: Check the file title here against what was added

An Environment ID is needed to configure and secure your application. Here’s how to get it:

* **Open the Developer Section**:  
   * Look at the menu on the left side of the screen. Find and click **Developers** to expand the options.  
* **Go to SDK & API Keys**:  
   * Under **Developers**, click on **SDK & API Keys**. This is where your Environment ID is stored.  
* **Copy the Environment ID**:  
   * Find the box labeled **Environment ID**. Click the copy icon next to the ID to copy it to your clipboard.

</Step>

   <Step title="Set Up Environment Variables">
   Create a `.env.local` file in the project’s root directory to store environment variables.

> ```

Reviewer: ```suggestion
```bash
```

mv .env.local.example .env.local
```

Setting up the `.env.local` file is critical for securely storing your environment ID. This ID is necessary for accessing Dynamic’s features and connecting to the Web3 backend.

Open the `.env.local` file and add your environment ID for Dynamic.

> ```

Reviewer: ```suggestion
```bash
```

NEXT_PUBLIC_DYNAMIC_ENVIRONMENT_ID=YOUR_ENVIRONMENT_ID
```
  </Step>
   <Step title="Run the Development Server">
    Start the development server using Bun or Yarn.
    <Tabs>
  <TabItem value="contribute" label="Using Bun" default>
    ```
    bun dev
    ```
  </TabItem>
  <TabItem value="contest" label="Using Yarn">
   ```
   yarn dev
   ```
  </TabItem>

</Tabs>

Visit [http://localhost:3000](http://localhost:3000) in your browser to view your project.

<img src="/img/developers/quickstart/dynamic-starter-kit/image2.png" alt="The homepage of the Rootstock Dynamic Starterkit"/>
  </Step>

</Steps>

## **Interacting with the Frontend**

<Steps>
  <Step title="Connect Wallet">
  * Use the `DynamicWidget` component for connecting to a wallet through options like MetaMask or WalletConnect. You can also offer social login options for enhanced accessibility.  

>    * Once logged in you will see a similar image like this

Reviewer: ```suggestion
   * Once logged in you will see a similar image like this:
```

<img src="/img/developers/quickstart/dynamic-starter-kit/image3.png" alt="Connect wallet using DynamicWidget component"/>

  </Step>
  <Step title="Check Token Balances">
   

>    The [`Balances`](https://github.com/RookieCol/rootstock-dynamic/blob/main/components/Balances.tsx) component fetches and displays the wallet's token balances, supporting multiple tokens like rBTC, tRIF, and DOC. 

Reviewer: ```suggestion
   The [`Balances`](https://github.com/RookieCol/rootstock-dynamic/blob/main/components/Balances.tsx) component fetches and displays the wallet's token balances, supporting multiple tokens like RBTC, tRIF, and DOC. 
```

  </Step>
  <Step title="Send Tokens">
   Through the [`Transfer`](https://github.com/RookieCol/rootstock-dynamic/blob/main/components/Transfer.tsx) component, users can transfer tokens directly within the dApp. It includes fields to specify the recipient address and token amount, along with secure hooks to initiate the transfer.

   **Features:**

>    * **Dropdown**: Select a token from available options (**rBTC, tRIF, and DOC**)  

Reviewer: ```suggestion
   * **Dropdown**: Select a token from available options (**RBTC, tRIF, and DOC**)  
```

   * **Input Fields**:

      - **Amount:** Enter the amount to send.

      - **Recipient Address:** Enter the address to send tokens to.
  </Step>

  <Step title="Sign Messages">
   The [`SignMessage`](https://github.com/RookieCol/rootstock-dynamic/blob/main/components/SignMessage.tsx) component enables the user to sign arbitrary messages using the connected wallet. This feature is useful for activities like authentication or data validation.

  </Step>
</Steps>
<br/>

> The **Dynamic Starter Kit for Rootstock** helps developers quickly add Web3 features to a **Next.js** app. Using **wagmi** hooks, it makes it easy to connect wallets, view token balances, transfer tokens, and sign messages, all within the app.

Reviewer: This doesn't resemble a conclusion, sounds like a paragraph for the introductory section. Summarize what was done in the guide than re-emphasizing what Dynamic is.
